### PR TITLE
simplify some NettyTransport code

### PR DESCRIPTION
**Pull request by pjfanning**
relates to #638

### CODE REVIEW

**Review by He-Pin — APPROVED** [review]
lgtm

File: remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala [hunk 1]
```diff
@@ -481,7 +481,7 @@ class NettyTransport(val settings: NettyTransportSettings, val system: ExtendedA
     bootstrap
   }
 
-  private val inboundBootstrap: Bootstrap = {
+  private val inboundBootstrap: ServerBootstrap = {
```
Comment (He-Pin) [inline remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala:484]: Need a mima filter for this

**Review by He-Pin — CHANGES_REQUESTED** [review]
Need update the mima filter

File: remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala (hunk 1 above)
Comment (pjfanning) [reply remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala:471]: I've fixed the issue - it was an import that needed removing, not a mima issue

The link validation failure - I have a fix in #640

**Review by He-Pin — APPROVED** [review]
Lgtm

**Review by He-Pin — APPROVED** [review]
Lgtm

**Pull request merged**